File: PostgreSQLVisual/Models/SensorData.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace PostgreSQLVisual.Models

{
    [Table("sens00_view", Schema = "public")] // Ensure this matches the actual table name in your PostgreSQL database
    public class SensorData
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.None)] // Since views don't auto-generate keys
        [Column("id")] // Ensure this matches the exact case as in the database
        public long Id { get; set; }

        [Column("mt_name")]
        [MaxLength(255)]
        public string MtName { get; set; }

        [Column("mt_value")]
        [MaxLength(255)]
        public string MtValue { get; set; }

        [Column("mt_time")]
        public DateTime MtTime { get; set; }

        [Column("mt_quality")]
        [MaxLength(1)]
        public string MtQuality { get; set; }

        // Additional properties or navigation properties for relationships can be added here
    }
}

File: backend/app/models/sensor_data.py
from backend.app import db
from sqlalchemy import Column, Integer, String, Float, DateTime

class SensorData(db.Model):
    """
    Example model that maps to your 'sens00' table in PostgreSQL.
    Adjust column names/types as needed to match your actual schema.
    """
    __tablename__ = 'sens00'  # Make sure this matches your actual table name
    
    mt_name = Column(String, primary_key=True)
    mt_time = Column(DateTime, primary_key=True)
    mt_value = Column(String)
    mt_quality = Column(String)                  # If your table has this column

    def __repr__(self):
        return f"<SensorDate(mt_time={self.mt_time}, mt_name={self.mt_name}, mt_value={self.mt_value})>"
